akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/ActorLoggingSpec.scala

```diff
@@ -0,0 +1,201 @@
+      }, occurrences = 5).intercept(
+        spawn(Behaviors.deferred[Any] { ctx ⇒
+          ctx.log.debug(marker, "whatever")
+          ctx.log.info(marker, "whatever")
+          ctx.log.warning(marker, "whatever")
+          ctx.log.error(marker, "whatever")
+          ctx.log.error(marker, cause, "whatever")
+          Behaviors.stopped
+        })
+      )
+    }
+
+    "provide a whole bunch of logging overloads" in {
+
+      // Not the best test but at least it exercises every log overload ;)
+
+      EventFilter.custom({
+        case _ ⇒ true // any is fine, we're just after the right count of statements reaching the listener
+      }, occurrences = 60).intercept {
+        spawn(Behaviors.deferred[String] { ctx ⇒
+          ctx.log.debug("message")
+          ctx.log.debug("{}", "arg1")
+          ctx.log.debug("{} {}", "arg1", "arg2")
+          ctx.log.debug("{} {} {}", "arg1", "arg2", "arg3")
+          ctx.log.debug("{} {} {} {}", "arg1", "arg2", "arg3", "arg4")
+          ctx.log.debug("{} {} {} {} {}", Array("arg1", "arg2", "arg3", "arg4", "arg5"))
+          ctx.log.debug(marker, "message")
+          ctx.log.debug(marker, "{}", "arg1")
+          ctx.log.debug(marker, "{} {}", "arg1", "arg2")
+          ctx.log.debug(marker, "{} {} {}", "arg1", "arg2", "arg3")
+          ctx.log.debug(marker, "{} {} {} {}", "arg1", "arg2", "arg3", "arg4")
+          ctx.log.debug(marker, "{} {} {} {} {}", Array("arg1", "arg2", "arg3", "arg4", "arg5"))
+
+          ctx.log.info("message")
+          ctx.log.info("{}", "arg1")
+          ctx.log.info("{} {}", "arg1", "arg2")
+          ctx.log.info("{} {} {}", "arg1", "arg2", "arg3")
+          ctx.log.info("{} {} {} {}", "arg1", "arg2", "arg3", "arg4")
+          ctx.log.info("{} {} {} {} {}", Array("arg1", "arg2", "arg3", "arg4", "arg5"))
+          ctx.log.info(marker, "message")
+          ctx.log.info(marker, "{}", "arg1")
+          ctx.log.info(marker, "{} {}", "arg1", "arg2")
+          ctx.log.info(marker, "{} {} {}", "arg1", "arg2", "arg3")
+          ctx.log.info(marker, "{} {} {} {}", "arg1", "arg2", "arg3", "arg4")
+          ctx.log.info(marker, "{} {} {} {} {}", Array("arg1", "arg2", "arg3", "arg4", "arg5"))
+
+          ctx.log.warning("message")
+          ctx.log.warning("{}", "arg1")
+          ctx.log.warning("{} {}", "arg1", "arg2")
+          ctx.log.warning("{} {} {}", "arg1", "arg2", "arg3")
+          ctx.log.warning("{} {} {} {}", "arg1", "arg2", "arg3", "arg4")
+          ctx.log.warning("{} {} {} {} {}", Array("arg1", "arg2", "arg3", "arg4", "arg5"))
+          ctx.log.warning(marker, "message")
+          ctx.log.warning(marker, "{}", "arg1")
+          ctx.log.warning(marker, "{} {}", "arg1", "arg2")
+          ctx.log.warning(marker, "{} {} {}", "arg1", "arg2", "arg3")
+          ctx.log.warning(marker, "{} {} {} {}", "arg1", "arg2", "arg3", "arg4")
+          ctx.log.warning(marker, "{} {} {} {} {}", Array("arg1", "arg2", "arg3", "arg4", "arg5"))
```

Comment:
I think that while we're at it we should fix another thing that comes up every so often: the ability to add an exception stack to warning log statements. That's about the only thing I'd change in this PR I think

Reply:
If you look in the Logger trait I noted that I don't think we can do that without breaking bincomp (because we'd have to either change or introduce new warning log events. But maybe I'm missing something?

Reply:
It's doable, I'll PR into your PR

Reply:
Done in https://github.com/johanandren/akka/pull/5 you can merge it into your PR